x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/container/ScoreSort.java

```diff
@@ -8,13 +8,22 @@
 import java.util.Objects;
 
 public class ScoreSort extends Sort {
-    public ScoreSort(Direction direction, Missing missing) {
+
+    final String id;
```

Comment:
Did you explore the idea of having the `id` as part of `Sort`, as I'm seeing a lot of repeated code? (skimming through this PR's changes, the `id` seems to be introduced in all classes inheriting `Sort`). I am probably missing something that prevents this approach to be used, would love to hear the reasons.

Reply:
I decided not to, because 2 classes the `AttributeSort` and `AggregateSort` can extract the id from their attribute variables. On the other hand 3 classes need to store it, I just chose the 1st approach with forcing to implement the `public String id()` getter.

Reply:
I would have done it differently. I feel like `id` and `id()` should belong to the same base class - `Sort`, with the help of an additional constructor (that receives the id as an argument). `Sort` would have also provided a default implementation for `id()` (to return the `id`) and the other classes would have overriden the default implementation of `id()` where appropriate. Also, these "special" cases classes would have called the aforementioned freshly added constructor in `Sort`.

The main bothering aspect for me is that there is a required `id()`  method in `Sort`, but the `id` itself (that _could_ be related to this `id()` method) lives in the inheriting classes.

You can leave it as is, I just wanted to point out something that doesn't feel right to _me_.